### PR TITLE
Update: Install typescript for navi-reports

### LINKED ISSUE #775: Install typescript navi reports

### PR DESCRIPTION
**Pull request by kevinhinterlong**
Resolves #775 

## Description
Install typescript

## Proposed Changes
- Small eslint changes

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/addon/components/navi-email-input.js [hunk 1]
```diff
@@ -25,7 +25,9 @@ export default Component.extend({
   /**
    * @property {Array} emails
    */
-  emails: computed(() => []),
+  emails: computed(function() {
+    return [];
+  }),
```
Comment (balajimanoharan) [inline packages/reports/addon/components/navi-email-input.js:30]:
```suggestion
  emails: [],
```

File: packages/reports/addon/components/power-select-collection-options.js [hunk 2]
```diff
@@ -46,7 +46,9 @@ export default Options.extend({
   /**
    * @property {Array} colums - columns sized using percentage widths
    */
-  columns: computed(() => [100]),
+  columns: computed(function() {
+    return [100];
+  }),
```
Comment (balajimanoharan) [inline packages/reports/addon/components/power-select-collection-options.js:51]:
```suggestion
  columns: [100],
```

File: packages/reports/addon/services/delivery-rule-action-dispatcher.js [hunk 3]
```diff
@@ -18,5 +18,7 @@ export default ActionDispatcher.extend({
   /**
    * @property {Array} consumers
    */
-  consumers: computed(() => ['delivery-rule'])
+  consumers: computed(function() {
+    return ['delivery-rule'];
+  })
```
Comment (balajimanoharan) [inline packages/reports/addon/services/delivery-rule-action-dispatcher.js:23]:
```suggestion
  consumers: ['delivery-rule']
```

File: packages/reports/addon/services/item-action-dispatcher.js [hunk 4]
```diff
@@ -16,5 +16,7 @@ export default ActionDispatcher.extend({
   /**
    * @property {Array} consumers
    */
-  consumers: computed(() => ['item'])
+  consumers: computed(function() {
+    return ['item'];
+  })
```
Comment (balajimanoharan) [inline packages/reports/addon/services/item-action-dispatcher.js:21]:
```suggestion
  consumers: ['item']
```

File: packages/reports/addon/services/request-action-dispatcher.js [hunk 5]
```diff
@@ -45,12 +45,14 @@ export default ActionDispatcher.extend({
...
+  consumers: computed(function() {
+    return [
+      'request/dimension',
+      'request/filter',
+      'request/logical-table',
+      'request/metric',
+      'request/time-grain',
+      'request/sort'
+    ];
+  })
```
Comment (balajimanoharan) [inline packages/reports/addon/services/request-action-dispatcher.js:57]:
```suggestion
  consumers: [
    'request/dimension',
    'request/filter',
    'request/logical-table',
    'request/metric',
    'request/time-grain',
    'request/sort'
  ]
```

File: packages/reports/addon/components/navi-email-input.js (hunk 1 above)
Comment (mattgraham1995) [reply packages/reports/addon/components/navi-email-input.js:30]: I think he'd have to make this into a native class to do that

Comment (kevinhinterlong) [reply packages/reports/addon/components/navi-email-input.js:30]: Yup, gonna close these since they're not native classes

**Pull request merged**